test/core/request_list.test.ts

```diff
@@ -780,7 +780,7 @@ describe('RequestList', () => {
 
             const name: string = null;
             const sources = [{ url: 'https://example.com' }];
-            const requests = sources.map(({ url }) => new Request({ url }));
+            const requests = sources.map(({ url }) => ({ url, uniqueKey: url }));
```

Comment:
Are you changing this so that it passes the equality check? Also, it'd be nice to call that variable `expectedRequests` or something.

Reply:
`new Request` creates the unique key in the constructor